src/symbols.cpp

```diff
@@ -0,0 +1,25 @@
+/*
+ * Copyright The async-profiler authors
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+#include "log.h"
+#include "symbols.h"
+
+void UnloadProtection::patchImport(ImportId id, void* hook_func) const {
```

Comment:
On one hand, I like your idea to protect `patchImport` calls on the interface level.
But at the same time, I'm not fan of blurring logic between CodeCache and Symbols. Things being equal, I'd prefer to minimize changes in existing code.

Reply:
70c3800772e488db3d9845853514fe6f1778e398